vertx-web/src/main/java/io/vertx/ext/web/handler/OtpAuthHandler.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.web.handler;
+
+import io.vertx.codegen.annotations.Fluent;
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.Handler;
+import io.vertx.ext.auth.otp.OtpKeyGenerator;
+import io.vertx.ext.auth.otp.hotp.HotpAuth;
+import io.vertx.ext.auth.otp.totp.TotpAuth;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.impl.HotpAuthHandlerImpl;
+import io.vertx.ext.web.handler.impl.TotpAuthHandlerImpl;
+
+/**
+ * An auth handler that provides One Time Password (Multi-Factor) Authentication support.
+ *
+ * @author Paulo Lopes
+ */
+@VertxGen
+public interface OtpAuthHandler extends Handler<RoutingContext> {
+
+  /**
+   * Create a new instance of this handler using a time based one time password authentication provider.
+   * @param totpAuth the time based OTP provider.
+   * @return new instance of the handler.
+   */
+  static OtpAuthHandler create(TotpAuth totpAuth) {
+    return new TotpAuthHandlerImpl(totpAuth, OtpKeyGenerator.create());
+  }
+
+  /**
+   * Create a new instance of this handler using a hash based one time password authentication provider.
+   * @param hotpAuth the hash based OTP provider.
+   * @return new instance of the handler.
+   */
+  static OtpAuthHandler create(HotpAuth hotpAuth) {
+    return new HotpAuthHandlerImpl(hotpAuth, OtpKeyGenerator.create());
+  }
+
+  /**
+   * Specify the URL where requests are to be redirected when a user is already known in the request.
+   *
+   * A user is already known when the {@link RoutingContext#user()} is not {@code null}.
+   *
+   * If no redirect is provided, no redirect will be done and requests are terminated immediatly with status code {@code 401}.
```

Comment:
```suggestion
   * If no redirect is provided, requests are terminated immediatly with status code {@code 401}.
```

I think this is a bit redundant as immediately implies that nothing else will happen in between